### PR TITLE
Azure container registries bugfixes and enhancements

### LINKED ISSUE #266: Expanding empty container registry throws errors
create a new azure container registry using the portal
browse to it in the explorer tree
when expanding an empty registry, it errors...

### PR DESCRIPTION
**Pull request by estebanreyl**
-Fixes #266 
- Improved azure explorer speed dramatically (6.5x +) by using an updated concurrent fetch request model that limits requests to 8 at a time. This was also implemented for images themselves. Note that while this is slower than my previous nonblocking approach it will avoid maintaining an unreasonable number of open requests. Thanks to @julialieberman for an idea regarding how to approach this.

- Alphabetically organized registries listed from azure and organized tags by date of creation.

- Added Microsoft/master rebase for easy merging

-Missed a , when merging, fixed

### CODE REVIEW

**Comment by StephenWeatherford** [comment]
I agree on that.  I just submitted a (PR)[#300] to turn on automatic formatting on save, which will keep this from happening again.  In the meantime, you should just do a Format Document on the files.

**Comment by StephenWeatherford** [comment]
Are you using VS to edit, and is that why there are so many formatting changes?  Recommended would be VS Code.

**Comment by estebanreyl** [comment]
Actually I think many have to do with with me downloading a beautify extension. Although the var x : number thing is definitely personal preference. I a happy to adapt though, how should I approach getting the style changes back to the desired style? @StephenWeatherford  mentioned a Format Document? Additionally is there any style guide that's being followed, I'd be happy to give it a read.

**Comment by StephenWeatherford** [comment]
Are you using VS Code?  SHIFT+ALT+F.  If you pull and merge my commit, that will happen automatically on every save.

**Comment by estebanreyl** [comment]
Formatting is now fixed

File: explorer/models/azureRegistryNodes.ts [hunk 1]
```diff
@@ -200,25 +200,34 @@ export class AzureRepositoryNode extends NodeBase {
...
-                node.password = element.password;
-                node.registry = element.registry;
-                node.serverUrl = element.repository;
-                node.subscription = element.subscription;
-                node.userName = element.userName;
-                node.created = moment(new Date(JSON.parse(manifest.history[0].v1Compatibility).created)).fromNow();
-                imageNodes.push(node);
-
+                pool.addTask(async () => {
+                    let data = await request.get('https://' + element.repository + '/v2/' + element.label + `/manifests/${tags[i]}`, {
```
[the remark this answers is not in the document]
Comment (estebanreyl) [reply explorer/models/azureRegistryNodes.ts]: That one addresses issue #256 . Not sure if I can somehow link it.

[the remark this answers is not in the document]
Comment (StephenWeatherford) [reply explorer/models/azureRegistryNodes.ts]: You mean 266?  Put "fixes #266" into the description.

**Comment by estebanreyl** [comment]
I've updated e branch according to your feedback and provided unit testing among other things for asyncpool. Let me know if there is anything else.

File: test/test.test.ts [hunk 2]
```diff
@@ -5,9 +5,135 @@
...
+        await pool.runAll();
+        assert.equal(counter, 100000);
+    })
+
+    test("Timer, Random 1-10 ms tests", async () => {
+        let pool = new AsyncPool(8);
+        let counter = 0;
+        for (let i = 0; i < 2000; i++) {
+            pool.addTask(async () => {
+                await sleep(Math.random() * 10);
```
Comment (StephenWeatherford) [inline test/test.test.ts]: Good!

**Comment by estebanreyl** [comment]
Thanks for all the feedback ! I've just updated the unit tests to take around 2 seconds.

**Pull request merged**